CI Flake `test_simple_close_delay_broadcast`

```
FAILED tests/test_closing.py::test_simple_close_delay_broadcast - TimeoutError: Unable to find "[re.compile('Resolved FUNDING_TRANSACTION/FUNDING_OUTPUT by MUTUAL_CLOSE')]" in logs.
```

```
lightningd-1 2026-07-06T14:43:51.598Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-simpleclosed-chan#1: Simple close: requesting delayed broadcast (our output 396625sat < peer 600000sat, our fee 3375sat < their fee 6750sat)
lightningd-1 2026-07-06T14:43:51.600Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-simpleclosed-chan#1: Status closed, but not exited. Killing
lightningd-1 2026-07-06T14:43:51.615Z INFO    033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-chan#1: State changed from CLOSINGD_SIGEXCHANGE to CLOSINGD_COMPLETE
lightningd-2 2026-07-06T14:43:51.624Z INFO    038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-chan#1: Simple close: stored closer tx 17a7fa9946b9972f6039c4bb35eb447eed840802abe8eff7ea73a85dafdeac0c
lightningd-2 2026-07-06T14:43:51.635Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-simpleclosed-chan#1: Closer tx stored by master: 17a7fa9946b9972f6039c4bb35eb447eed840802abe8eff7ea73a85dafdeac0c
lightningd-2 2026-07-06T14:43:51.637Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-simpleclosed-chan#1: Status closed, but not exited. Killing
lightningd-2 2026-07-06T14:43:51.651Z INFO    038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-chan#1: State changed from CLOSINGD_SIGEXCHANGE to CLOSINGD_COMPLETE
lightningd-1 2026-07-06T14:43:51.657Z INFO    033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-chan#1: Simple close: delaying broadcast by 1 hour (peer has higher-fee tx)
lightningd-1 2026-07-06T14:43:51.678Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-connectd: peer_downgrade
lightningd-2 2026-07-06T14:43:51.818Z DEBUG   wallet: Owning output 1 593250sat (p2tr) txid 17a7fa9946b9972f6039c4bb35eb447eed840802abe8eff7ea73a85dafdeac0c
lightningd-2 2026-07-06T14:43:51.832Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-chan#1: We have 2 anchor points to use
lightningd-2 2026-07-06T14:43:51.836Z DEBUG   lightningd: Broadcasting txid 17a7fa9946b9972f6039c4bb35eb447eed840802abe8eff7ea73a85dafdeac0c
lightningd-2 2026-07-06T14:43:51.839Z DEBUG   lightningd: sendrawtransaction: 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000
lightningd-2 2026-07-06T14:43:51.843Z DEBUG   hsmd: Client: Received message 5 from client
lightningd-2 2026-07-06T14:43:51.857Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-connectd: peer_downgrade
lightningd-2 2026-07-06T14:43:52.077Z DEBUG   plugin-bcli: sendrawtx exit 0 (bitcoin-cli -regtest -datadir=/tmp/ltests-8o9aeg9a/test_simple_close_delay_broadcast_1/lightning-2/ -rpcclienttimeout=60 -rpcport=59913 -rpcuser=... -stdinrpcpass -stdin sendrawtransaction 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000) 
lightningd-2 2026-07-06T14:43:52.082Z DEBUG   lightningd: Adding block 104: 0ef57345d58b086b13dd9c387f73e26ff19b57d2e04f0b2130ad90d8c48a0599
lightningd-2 2026-07-06T14:43:52.116Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-chan#1: Funding tx a8529ab193184808a71720ae22e56b65ec04da2f7805ab1546db430a5c8b95d6 depth 2 of 1
lightningd-2 2026-07-06T14:43:52.116Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-chan#1: Funding tx a8529ab193184808a71720ae22e56b65ec04da2f7805ab1546db430a5c8b95d6 confirmed, but peer in state CLOSINGD_COMPLETE
lightningd-2 2026-07-06T14:43:52.120Z DEBUG   lightningd: sendrawtransaction: 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000
lightningd-2 2026-07-06T14:43:52.126Z DEBUG   gossipd: REPLY WIRE_GOSSIPD_NEW_BLOCKHEIGHT_REPLY with 0 fds
lightningd-2 2026-07-06T14:43:52.154Z DEBUG   plugin-bcli: sendrawtx exit 0 (bitcoin-cli -regtest -datadir=/tmp/ltests-8o9aeg9a/test_simple_close_delay_broadcast_1/lightning-2/ -rpcclienttimeout=60 -rpcport=59913 -rpcuser=... -stdinrpcpass -stdin sendrawtransaction 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000) 
lightningd-1 2026-07-06T14:43:52.835Z DEBUG   lightningd: Adding block 104: 0ef57345d58b086b13dd9c387f73e26ff19b57d2e04f0b2130ad90d8c48a0599
lightningd-1 2026-07-06T14:43:52.855Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-chan#1: Funding tx a8529ab193184808a71720ae22e56b65ec04da2f7805ab1546db430a5c8b95d6 depth 2 of 1
lightningd-1 2026-07-06T14:43:52.855Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-chan#1: Funding tx a8529ab193184808a71720ae22e56b65ec04da2f7805ab1546db430a5c8b95d6 confirmed, but peer in state CLOSINGD_COMPLETE
lightningd-1 2026-07-06T14:43:52.862Z DEBUG   gossipd: REPLY WIRE_GOSSIPD_NEW_BLOCKHEIGHT_REPLY with 0 fds
lightningd-1 2026-07-06T14:43:58.155Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-connectd: peer_out WIRE_GOSSIP_TIMESTAMP_FILTER
lightningd-2 2026-07-06T14:43:58.155Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-connectd: peer_in WIRE_GOSSIP_TIMESTAMP_FILTER
lightningd-1 2026-07-06T14:43:58.155Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-gossipd: seeker: chosen for periodic full sync
lightningd-1 2026-07-06T14:43:58.155Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-gossipd: seeker: starting gossip (EVERYTHING)
lightningd-2 2026-07-06T14:43:58.178Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-gossipd: seeker: chosen for periodic full sync
lightningd-2 2026-07-06T14:43:58.178Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-gossipd: seeker: starting gossip (EVERYTHING)
lightningd-2 2026-07-06T14:43:58.179Z DEBUG   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-connectd: peer_out WIRE_GOSSIP_TIMESTAMP_FILTER
lightningd-1 2026-07-06T14:43:58.188Z DEBUG   033845802d25b4e074ccfd7cd8b339a41dc75bf9978a034800444b51d42b07799a-connectd: peer_in WIRE_GOSSIP_TIMESTAMP_FILTER
lightningd-2 2026-07-06T14:44:39.164Z DEBUG   lightningd: sendrawtransaction: 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000
lightningd-2 2026-07-06T14:44:39.249Z DEBUG   plugin-bcli: sendrawtx exit 0 (bitcoin-cli -regtest -datadir=/tmp/ltests-8o9aeg9a/test_simple_close_delay_broadcast_1/lightning-2/ -rpcclienttimeout=60 -rpcport=59913 -rpcuser=... -stdinrpcpass -stdin sendrawtransaction 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000) 
lightningd-2 2026-07-06T14:45:07.169Z TRACE   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-lightningd: Ping latency: 10083413nsec
lightningd-2 2026-07-06T14:45:36.256Z DEBUG   lightningd: sendrawtransaction: 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000
lightningd-2 2026-07-06T14:45:36.429Z DEBUG   plugin-bcli: sendrawtx exit 0 (bitcoin-cli -regtest -datadir=/tmp/ltests-8o9aeg9a/test_simple_close_delay_broadcast_1/lightning-2/ -rpcclienttimeout=60 -rpcport=59913 -rpcuser=... -stdinrpcpass -stdin sendrawtransaction 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000) 
lightningd-2 2026-07-06T14:46:12.165Z TRACE   038194b5f32bdf0aa59812c86c4ef7ad2f294104fa027d1ace9b469bb6f88cf37b-lightningd: Ping latency: 6996283nsec
lightningd-2 2026-07-06T14:46:34.433Z DEBUG   lightningd: sendrawtransaction: 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000
lightningd-2 2026-07-06T14:46:34.503Z DEBUG   plugin-bcli: sendrawtx exit 0 (bitcoin-cli -regtest -datadir=/tmp/ltests-8o9aeg9a/test_simple_close_delay_broadcast_1/lightning-2/ -rpcclienttimeout=60 -rpcport=59913 -rpcuser=... -stdinrpcpass -stdin sendrawtransaction 02000000000101d6958b5c0a43db4615ab05782fda04ec656be522ae2017a708481893b19a52a80000000000fdffffff02801a06000000000022512028d40955f64f5a5a502ae8ecdf3fe125df90f112fc87b53657d54f1b012943ea620d090000000000225120d2b3d07ad1463778f5835037e329a0ea69fdfed6d6fe54f84599a9e0cac98b9e0400473044022064a66bbce07d71b84f536dac817151424c3f7d9015832fadc04e8f461c4593e10220497cdbe2e21aab83e5fdbd3106d73b3a81ec56da4b86771311be9c37b24dde8401473044022040085b15a56cf61c0bc57ac785e23d09aaf62f63234f0e4bea95d1f98e7d7b5a02204b22806f53bdd3e967ff98d26519c86d8f412ead54e3830846b0f4a6f8b6e9e20147522102096e74aff8b917448bf683a9e60058194a77bfea42360f8ad2434763dd63527c21025b93ec4849eec35a35063a7e449c22a85940ececea6d0f40723f7b5f5afc9b6852ae00000000) 
Time-out: can't find [re.compile('Resolved FUNDING_TRANSACTION/FUNDING_OUTPUT by MUTUAL_CLOSE')] in logs
```


Full example:
https://github.com/ElementsProject/lightning/actions/runs/28798310878/job/85399257273?pr=9275

--- a/tests/test_closing.py
+++ b/tests/test_closing.py
@@ -4358,11 +4358,14 @@ def test_simple_close_delay_broadcast(node_factory, bitcoind, executor):
              == 'CLOSINGD_COMPLETE')
     assert not l2.daemon.is_in_log('Simple close: delaying broadcast')
 
-    # l2 broadcasts immediately; wait until its tx is confirmed.
-    # We wait for the broadcast log rather than polling getrawmempool(),
-    # which can miss a just-submitted tx under the rpcproxy timing.
+    # l2 broadcasts immediately; wait until its tx is actually in the mempool
+    # before mining.  The 'Broadcasting txid' log only means CLN *called*
+    # sendrawtransaction, not that bitcoind accepted the tx: mining on the log
+    # alone can produce an empty block under rpcproxy timing, leaving the
+    # funding output unspent so MUTUAL_CLOSE never resolves.  generate_block's
+    # wait_for_mempool polls until the tx appears, so it can't miss it.
     l2.daemon.wait_for_log('Broadcasting txid')
-    bitcoind.generate_block(1)
+    bitcoind.generate_block(1, wait_for_mempool=1)
     l1.daemon.wait_for_log('Resolved FUNDING_TRANSACTION/FUNDING_OUTPUT by MUTUAL_CLOSE')
     l2.daemon.wait_for_log('Resolved FUNDING_TRANSACTION/FUNDING_OUTPUT by MUTUAL_CLOSE')
     fut.result(timeout=10)

